### PR TITLE
feat: share post, create message, get messages // fix: fixed swagget …

### PR DESCRIPTION
**Pull request by gabrikf**
feat: 
Created Routes, Controllers, Use Cases, Dtos, Unit tests and e2e tests for:
 - Create message.
 - Get conversation.
 - Share post with multiple users.
 fix: 
 - Swagger
 - Tests

### CODE REVIEW

File: src/modules/users/infra/prisma/repositories/MessagesRepository.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            (item) =>
+              item.fromUserId === value.fromUserId ||
+              item.toUserId === value.toUserId
+          )
+        );
+      }
+    );
+    return filterDuplicates;
+  }
+  // async getAllFromUser({
```
Comment (brunormueller) [inline src/modules/users/infra/prisma/repositories/MessagesRepository.ts:60]: Let's remove this, man?

File: src/modules/users/useCases/createMessage/CreateMessageController.spec.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import { app } from "@shared/infra/http/app";
+import { IUserForTest, user } from "@shared/infra/jest/jest.setup";
+import { randomUUID } from "crypto";
+import request from "supertest";
+
+describe("CreateMessageController", () => {
+  const user2: IUserForTest = {
+    full_name: "John Doe",
```
Comment (brunormueller) [inline src/modules/users/useCases/createMessage/CreateMessageController.spec.ts:8]: John Doe is famous in ur country? Every file i saw that guy 😢

**Review by brunormueller — COMMENTED** [review]
Fabulous job Gabe!
Do u did a lot, appreciate that!

File: src/modules/users/infra/prisma/repositories/MessagesRepository.ts (hunk 1 above)
Comment (gabrikf) [reply src/modules/users/infra/prisma/repositories/MessagesRepository.ts]: fixed

File: src/modules/users/useCases/createMessage/CreateMessageUseCase.spec.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    userRepository = new UserRepositoryInMemory();
+    messagesRepository = new MessageRepositoryInMemory();
+    createUserUseCase = new CreateUserUseCase(userRepository);
+    createMessageUseCase = new CreateMessageUseCase(
+      messagesRepository,
+      userRepository
+    );
+    user = (
+      await createUserUseCase.execute({
+        full_name: "opateste",
```
[the remark this answers is not in the document]
Comment (gabrikf) [reply src/modules/users/useCases/createMessage/CreateMessageUseCase.spec.ts:28]: He is John's brother...

File: src/modules/users/useCases/createMessage/CreateMessageController.spec.ts (hunk 2 above)
Comment (gabrikf) [reply src/modules/users/useCases/createMessage/CreateMessageController.spec.ts:8]: He is a good man, everyone knows about him...

**Pull request merged**